src/cpp/src/image_generation/models/unet_inference.hpp

```diff
@@ -0,0 +1,68 @@
+// Copyright (C) 2023-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "openvino/genai/image_generation/unet2d_condition_model.hpp"
+
+namespace ov {
+namespace genai {
+
+class UNet2DConditionModel::UNetInference {
+
+public:
+    virtual void compile(std::shared_ptr<ov::Model> model, const std::string& device, const ov::AnyMap& properties) = 0;
+    virtual void set_hidden_states(const std::string& tensor_name, ov::Tensor encoder_hidden_states) = 0;
+    virtual void set_adapters(AdapterController& adapter_controller, const AdapterConfig& adapters) = 0;
+    virtual ov::Tensor infer(ov::Tensor sample, ov::Tensor timestep) = 0;
+
+    // utility function to resize model given optional dimensions.
+    static void reshape(std::shared_ptr<ov::Model> model,
+                        std::optional<int> batch_size = {},
+                        std::optional<int> height = {},
+                        std::optional<int> width = {},
+                        std::optional<int> tokenizer_model_max_length = {})
```

Comment:
why do we need them to be `optional` ? as I see from the calling code, you are specifying all the parameters anyway

Reply:
We call this function with *only* batch size, from here: https://github.com/openvinotoolkit/openvino.genai/blob/ed2baf4461c330de3c5a38cae124cbf859ab2bf5/src/cpp/src/image_generation/models/unet_inference_static_bs1.hpp#L43